### DISCUSSION TITLE
Add ?() operator

### DISCUSSION THREAD

**Discussion by tanukihee** [dotnet/csharplang#5724] [Language Ideas] [answered]
Now:
```C#
action?.Invoke(args);
```

Can be:
```C#
action?(args);
```

c.f. `obj.func?.(args)` in [JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

**Comment by CyrusNajmabadi** [comment, marked as answer]
This is a dupe of some issue.  the primary issues here are:

1. implementation complexity.  The above is already a legal prefix today of other expressions.  For example: `action ? (args) : etc`.
2. Because it's a legal prefix, this becomes highly challenging for *humans* to parse and recognize what is going on.  In other words, even if we can get the impl to do this properly, we still feel like it would be confusing for normal people to grok what is going on.

Because of both of those issues, it is highly unlikely to ever happen with that syntax.

**Reply by MgSam** [reply]
What about the JS syntax? I imagine they wrestled with the same issues before they decided on `?.()`

**Reply by CyrusNajmabadi** [reply]
That's a potential issue of we want suffix casts with the x.(type) syntax

**Reply by jnm2** [reply]
Suffix cast discussion: https://github.com/dotnet/csharplang/discussions/5447

**Comment by jnm2** [comment]
Existing discussion: https://github.com/dotnet/csharplang/discussions/4725
